### PR TITLE
Bug 78740 disallow radix prefix envvars

### PR DESCRIPTION
**Pull request by magnus-longva-bouvet**
Bug 78740

### CODE REVIEW

File: pkg/apis/radixvalidators/validate_ra.go [hunk 1]
```diff
@@ -834,9 +836,27 @@ func validateMaxNameLengthForAppAndEnv(appName, envName string) error {
...
+	if err := validateIllegalPrefixInVariableName(resourceName, value); err != nil {
+		return err
+	}
 	return validateResourceWithRegexp(resourceName, value, "^(([A-Za-z0-9][-._A-Za-z0-9.]*)?[A-Za-z0-9])?$")
 }
 
+func validateIllegalPrefixInVariableName(resourceName string, value string) error {
+	for _, illegalPrefix := range illegalVariableNamePrefixes {
+		var comparisonIndex = 0
+		if len(value) < len(illegalPrefix) {
```
Comment (satr) [inline pkg/apis/radixvalidators/validate_ra.go:848]: I think you need to validate `resourceName` (variable name) - not its value.

File: pkg/apis/radixvalidators/validate_ra.go [hunk 2]
```diff
@@ -834,9 +836,27 @@ func validateMaxNameLengthForAppAndEnv(appName, envName string) error {
...
 
 func validateVariableName(resourceName, value string) error {
+	if err := validateIllegalPrefixInVariableName(resourceName, value); err != nil {
+		return err
+	}
 	return validateResourceWithRegexp(resourceName, value, "^(([A-Za-z0-9][-._A-Za-z0-9.]*)?[A-Za-z0-9])?$")
 }
 
+func validateIllegalPrefixInVariableName(resourceName string, value string) error {
+	for _, illegalPrefix := range illegalVariableNamePrefixes {
```
Comment (satr) [inline pkg/apis/radixvalidators/validate_ra.go:846]: There is a method `IsRadixEnvVar` to check if variable name is with "that" prefixes

**Review by satr — CHANGES_REQUESTED** [review]

File: pkg/apis/radixvalidators/validate_ra.go (hunk 1 above)
Comment (magnus-longva-bouvet) [reply pkg/apis/radixvalidators/validate_ra.go:848]: Not quite sure what `resourceName` is in this context, but it shows up as "environment variable name" when debugging the tests. The `value` appears to be the name of the environment variables.

File: pkg/apis/radixvalidators/validate_ra.go [hunk 3]
```diff
@@ -834,9 +837,20 @@ func validateMaxNameLengthForAppAndEnv(appName, envName string) error {
...
+		return err
+	}
 	return validateResourceWithRegexp(resourceName, value, "^(([A-Za-z0-9][-._A-Za-z0-9.]*)?[A-Za-z0-9])?$")
 }
 
+func validateIllegalPrefixInVariableName(resourceName string, value string) error {
+	if utils.IsRadixEnvVar(value) {
+		return fmt.Errorf("%s %s can not start with prefix reserved for platform", resourceName, value)
+	} else {
+		return nil
```
Comment (satr) [inline pkg/apis/radixvalidators/validate_ra.go:850]: Not needed `else`

**Review by satr — APPROVED** [review]

**Pull request merged**